tests/main/document-interfaces-url/task.yaml

```diff
@@ -0,0 +1,84 @@
+summary: Ensure that the output of snap interface <interface-name> contains an existing url
+
+details: |
+    Each interface should contain a url pointing to a documentation page found on
+    snapcraft.io as https://snapcraft.io/docs/<interface-name>-interface. That page
+    must also exist.
+
+systems: [ubuntu-24.04-64]
+
+execute: |
+    # Checks to see if the url supplied as the first argument returns an OK HTML status code
+    webpage_exists() {
+        local url=$1
+        local status_code
+        local num_retries=3
+
+        for _ in $(seq $num_retries); do
```

Comment:
why do we need a retry here? have you seen that fail to get the status code?

Reply:
It did fail on me once, and I only ran it by itself a few times. So I figured it would fail pretty regularly and needed some extra help.